### PR TITLE
Improve overflow with OverflowItemFixed.Ellipsis

### PR DESCRIPTION
**Pull request by JamesNK**
Use the new OverflowItemFixed.Ellipsis feature.
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/4976)

### CODE REVIEW

File: Directory.Build.props [hunk 1]
```diff
@@ -16,7 +16,7 @@
     <PackageIconFullPath>$(SharedDir)Aspire_icon_256.png</PackageIconFullPath>
     <PackageProjectUrl>https://github.com/dotnet/aspire</PackageProjectUrl>
     <WarnOnPackingNonPackableProject>false</WarnOnPackingNonPackableProject>
-    <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
+    <TreatWarningsAsErrors>false</TreatWarningsAsErrors>
```
Comment (JamesNK) [inline Directory.Build.props:19]: Temporary hack to get Aspire to build. Preview FluentUI package is signed with an untrusted cert.

**Comment by JamesNK** [comment]
Bug: https://github.com/microsoft/fluentui-blazor/issues/2427

**Comment by JamesNK** [comment]
Bugs fixed. Please review.

File: src/Aspire.Dashboard/Components/Controls/Chart/ChartFilters.razor [hunk 2]
```diff
@@ -22,19 +22,25 @@
                                 }
                                 else
                                 {
+                                    var i = 0;
                                     foreach (var item in context.SelectedValues.OrderBy(g => g.Name))
                                     {
-                                        <FluentOverflowItem>
-                                            <FluentBadge>@item.Name</FluentBadge>
+                                        <FluentOverflowItem Fixed="@((i == 0) ? OverflowItemFixed.Ellipsis : OverflowItemFixed.None)">
```
Comment (drewnoakes) [inline src/Aspire.Dashboard/Components/Controls/Chart/ChartFilters.razor:28]: It's not clear to me why the first item is treated differently. Can we add a comment?

Comment (JamesNK) [inline src/Aspire.Dashboard/Components/Controls/Chart/ChartFilters.razor:28]:
```suggestion
                                        // Always display the first item by setting a fixed value.
                                        <FluentOverflowItem Fixed="@((i == 0) ? OverflowItemFixed.Ellipsis : OverflowItemFixed.None)">
```

File: src/Aspire.Dashboard/Components/ResourcesGridColumns/EndpointsColumnDisplay.razor [hunk 3]
```diff
@@ -14,7 +14,7 @@ else if (DisplayedEndpoints.Count > 1)
                 var displayedEndpoint = DisplayedEndpoints[i];
                 var isLast = i == DisplayedEndpoints.Count - 1;
 
-                <FluentOverflowItem Data="displayedEndpoint">
+                <FluentOverflowItem Data="displayedEndpoint" Fixed="@((i == 0) ? OverflowItemFixed.Ellipsis : OverflowItemFixed.None)">
```
Comment (JamesNK) [inline src/Aspire.Dashboard/Components/ResourcesGridColumns/EndpointsColumnDisplay.razor:17]:
```suggestion
                 // Always display the first item by setting a fixed value.
                <FluentOverflowItem Data="displayedEndpoint" Fixed="@((i == 0) ? OverflowItemFixed.Ellipsis : OverflowItemFixed.None)">
```

File: src/Aspire.Dashboard/Components/Controls/Chart/ChartFilters.razor (hunk 2 above)
Comment (JamesNK) [reply src/Aspire.Dashboard/Components/Controls/Chart/ChartFilters.razor:29]: Done

**Review by drewnoakes — APPROVED** [review]

**Pull request merged**